### DISCUSSION TITLE
SolarEdge data to SmartEVSE

### DISCUSSION THREAD

**Discussion by devdems** [serkri/SmartEVSE-3#214] [Q&A] [answered]
Hi, i have a SolarEdge inverter which has an external meter connected with Modbus.

I would like to get all those values to SmartEVSE. What are my options? 

Does this version of FW support Modbus over TCP as that would probably be the easiest way to do it because it is already supported in SolarEdge?

**Comment by dingo35** [comment]
You can configure RTU modbus meters through the  "Custom values" submenu, if you know all the register values and formats.

Tcp Modbus is only reading stuff off of the modbus via a bridge, putting values on the modbus is uncharted territory, would depend on what the Arduino eModbus library is supporting....

**Reply by devdems** [reply]
The problem is that SolarEdge is already master and the meter is slave. If i'm not mistaken, you can only have one master so i don't see a way to connect directly to meter's modbus to get the values. You were probably talking about this kind of connection, right?

**Comment by dingo35** [comment]
Yes you are supposed to have only one master on an RTU modbus, although there are people who run two masters, ignoring incidental timeouts/crc errors when collisions occur. YMMV.

Using your Solaredge as a PV meter in SmartEVSE would IMHO not add any value; measuring your mains import/export, which includes the usage of your house, makes more sense....

**Reply by devdems** [reply]
The meter (IneoPro Pro380) that is connected to SolarEdge measures all the information of the house. That is why i need this info in smartevse so it will be able to charge based on the house consumption and solar production.

**Comment by dingo35** [comment]
I see. Can you get the data of your meter into HomeAssistant?

**Reply by devdems** [reply]
I can, yes. I can also get it into MQTT.

**Comment by dingo35** [comment, marked as answer]
You can use both the REST API (https://swagger-ui.serkri.be/) or the MQTT interface (not well documented yet, see source code or tweakers.net) to feed the mains info to SmartEVSE.
